### PR TITLE
Add more plugins

### PR DESCRIPTION
**Pull request by Erick555**
This add unarchiver and p7zip plugins.

### CODE REVIEW

**Comment by Erick555** [comment]
bot, build

**Comment by Siosm** [comment]
Could you also add the security patches from https://github.com/archlinux/svntogit-packages/tree/packages/p7zip/trunk
Thanks for working on that!

File: org.kde.ark.json [hunk 1]
```diff
@@ -105,6 +105,81 @@
                 }
             ]
         },
+        {
+         "name": "gnustep-make",
+         "buildsystem": "simple",
```
Comment (Siosm) [inline org.kde.ark.json:110]: Have you tried `autotools` here?

File: org.kde.ark.json [hunk 2]
```diff
@@ -105,6 +105,81 @@
...
+            {
+               "type": "archive",
+               "url": "https://github.com/gnustep/tools-make/archive/make-2_8_0.tar.gz",
+               "sha256": "a9df9b0d756a93306a8e0cb9e127f5688a459dad3d356cd760565833da605825"
+            }
+         ]
+        },
+        {
+         "name": "gnustep",
+         "buildsystem": "simple",
```
Comment (Siosm) [inline org.kde.ark.json:125]: Have you tried autotools here?

File: org.kde.ark.json [hunk 3]
```diff
@@ -105,6 +105,94 @@
...
+         "name": "unarchiver",
+         "buildsystem": "simple",
+         "build-commands": [
+            "sed -s -i 's#/usr/include/GNUstep#/app/include#g' */Makefile.linux && cd XADMaster && make -j $FLATPAK_BUILDER_N_JOBS -f Makefile.linux",
+            "install -m755 XADMaster/unar XADMaster/lsar /app/bin/"
+         ],
+         "sources": [
+            {
+               "type": "archive",
+               "url": "https://cdn.theunarchiver.com/downloads/TheUnarchiverSource.zip",
```
Comment (Siosm) [inline org.kde.ark.json:149]: Arch Linux moved to https://github.com/MacPaw/XADMaster as upstream: https://github.com/archlinux/svntogit-community/blob/packages/unarchiver/trunk/PKGBUILD

**Review by Siosm — CHANGES_REQUESTED** [review]

**Comment by tsdgeos** [comment]
okular builds unarchiver if you want to get inspired by it

**Comment by tsdgeos** [comment]
or maybe we need similar fixes :D Even one of those shared json things

**Comment by Erick555** [comment]
@tsdgeos I literally copied unarrchiver part from okular as [stated in commit](https://github.com/flathub/org.kde.ark/pull/14/commits/3475a9e016ce193353ca6f12d53cb4c8aabf39b5).

File: org.kde.ark.json (hunk 1 above)
Comment (Erick555) [reply org.kde.ark.json:110]: All of this is copied from [okular](https://github.com/flathub/org.kde.okular/blob/master/org.kde.okular.json#L43) manifest which guarantees this works so I think if there is any improvement to make then it should be done in both (alternatively shared-module may be proposed). I'll prefer not doing that in this PR though.

**Comment by tsdgeos** [comment]
> @tsdgeos I literally copied unarrchiver part from okular as [stated in commit](https://github.com/flathub/org.kde.ark/pull/14/commits/3475a9e016ce193353ca6f12d53cb4c8aabf39b5).

Sorry for not reading ^_^

File: org.kde.ark.json (hunk 1 above)
Comment (tsdgeos) [reply org.kde.ark.json:110]: I'm with @Erick555 here, this is relatively complicated so i'd say let's land this, see if we can get it into shared-module and then do improvemetns?

File: org.kde.ark.json (hunk 3 above)
Comment (Siosm) [reply org.kde.ark.json:149]: I'll open an issue for this one.

**Review by Siosm — COMMENTED** [review]
LGTM. Will test soon but feel free to merge if you get to it before me.

**Review by Siosm — APPROVED** [review]

**Comment by Erick555** [comment]
@tsdgeos @travier ping, I tested that plugins are available in ark config and opening 7z archive works.

**Pull request merged**